### PR TITLE
sylpheed: become maintainer, update to 3.7.0, fix building on macOS 11 and newer

### PR DESCRIPTION
**Pull request by fincham**
#### Description

I'd like to become the maintainer for this port. I've attached two commits: firstly, just making myself the maintainer (and adding openmaintainer), and secondly to update to the upstream version 3.7.0 and add the usual libtool patch to fix building on recent macOS versions. 

I've got a few other tidy-ups I'd like to do to this package (like adding variant descriptions, making the SSL build default, cleaning up unused patches) which I'll action later. I've tested this build with both +x11 and +quartz versions of GTK2.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.0.1 21A559 arm64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: mail/sylpheed/Portfile [hunk 1]
```diff
@@ -3,12 +3,12 @@
 PortSystem          1.0
 
 name                sylpheed
-version             3.5.1
+version             3.7.0
 revision            1
```
Comment (reneeotten) [inline mail/sylpheed/Portfile:7]: when updating the `version` you should reset the `revision` to zero

**Comment by pmetzger** [comment]
Side note:, `claws-mail` is a fork of sylpheed; I stopped maintaining that a while ago when I ceased to use it.

**Comment by mascguy** [comment]
Michael, just started the CI jobs for you. And since this is your first contribution, we wish you a warm welcome from the entire MacPorts family!

File: mail/sylpheed/Portfile (hunk 1 above)
Comment (fincham) [reply mail/sylpheed/Portfile]: Fixed

**Comment by fincham** [comment]
Thank you for the kind welcome! I've just fixed the `revision` field and re-pushed this branch.

**Comment by reneeotten** [comment]
thanks @fincham if you want with a future update you could add a description to the variants (`port lint` warns about them missing) and instead of doing thing like:
```
configure.args-delete   --disable-ssl
configure.args-append   --enable-ssl
```

you could use : ```configure.args-replace   --disable-ssl --enable-ssl```, but these are all cosmetic changes and do not affect the functioning of the port.

**Pull request merged**